### DISCUSSION TITLE
'Cmd-c' shortcut doesn't work (on Mac) while 'Cmd-backspace' and 'C-c' does

### DISCUSSION THREAD

**Discussion by VasylBaran** [helix-editor/helix#12270] [Troubleshooting]
### Summary

I'm loving helix. While trying to customize the shortcuts a bit I've added the following to my config.toml:

`Cmd-c = "yank"`
`Cmd-v = "paste_after"`

In this configuration 'paste_after' works but 'yank' doesn't (nothing happens)

If I trigger 'yank' in a usual way (Space + y) it works, so it's more about the shortcut rather than about method.

Similar trouble with 'S-backspace' assigned to "delete_selection" - nothing happens when it's 'S-backspace' but 'Cmd-backspace' works.

### Reproduction Steps

Add the following to config.toml:

`Cmd-c = "yank"`
`Cmd-v = "paste_after"`

When I press "Cmd-c" (Command+C) I expected the selection to get copied (yanked). Instead, nothing happens.

### Platform

macOS

### Terminal Emulator

Alacritty

### Installation Method

source

### Helix Version

24.07 (e14c346ee74a44051b2c07c2255a6ab80142dbe7)

**Comment by VasylBaran** [comment]
@the-mikedavis thank you for adding support for Cmd key in 271c32f2e6ead6fea48b316947fcc574fe3865b5

I wonder, have you tried to assign any action to 'Cmd-c' shortcut inside config.toml file? Does it work for you and does it trigger the action?

Thanks again! ;-)

**Reply by the-mikedavis** [reply]
A `Cmd-c` binding works for me locally but I'm using Linux and Kitty. Either macOS or the terminal emulator might be eating that keypress (I would suspect macOS). Also note that super bindings (Cwd/Win/Meta) require a terminal that supports the enhanced keyboard protocol - I'm not sure off the top of my head whether Alacritty does. You can check by starting Helix with `hx -vv` and looking for a line like "[DEBUG] The keyboard enhancement protocol is supported in this terminal (checked in 3.140555ms)". (It will say "is not" if Alacritty doesn't support the enhanced keyboard protocol.)

**Reply by VasylBaran** [reply]
> A `Cmd-c` binding works for me locally but I'm using Linux and Kitty. Either macOS or the terminal emulator might be eating that keypress (I would suspect macOS). Also note that super bindings (Cwd/Win/Meta) require a terminal that supports the enhanced keyboard protocol - I'm not sure off the top of my head whether Alacritty does. You can check by starting Helix with `hx -vv` and looking for a line like "[DEBUG] The keyboard enhancement protocol is supported in this terminal (checked in 3.140555ms)". (It will say "is not" if Alacritty doesn't support the enhanced keyboard protocol.)

you're absolutely right! The same shortcut works in kitty, but not in Alacritty. Although I do see "helix_tui::backend::crossterm [DEBUG] The keyboard enhancement protocol is supported in this terminal" so theoretically it "should" work. Thank you for your insight @the-mikedavis!

**Reply by VasylBaran** [reply]
Cmd-c however doesn't work, not even in kitty =(

**Reply by VasylBaran** [reply]
Cmd-S-c works, Cmd-c doesn't

**Reply by astrob3rry** [reply]
I'm also experiencing this key remapping difficulty on Kitty (macOS): `Cmd-S-c` yanks, but `Cmd-c` doesn't.

**Reply by NathanielWroblewski** [reply]
I am experiencing difficulties mapping `Cmd+c` as well:
  - I am running `helix` through `wezterm` on `NixOS`.
  - I've set the line `enable_kitty_keyboard = true` in my `wezterm.lua` configuration
  - I've verified that the keyboard enhancement protocol is detected by opening helix with `hx -vv` and then executing `:log-open`, jumping to the bottom with `ge` and locating the following log entry:
  ```
  2025-07-07T00:09:16.141 helix_tui::backend::crossterm [DEBUG] The keyboard enhancement protocol is supported in this terminal (checked in 4.611393ms)
```
  - my helix configuration has the following entries:
```
[editor]
clipboard-provider = "wayland"
default-yank-register = "+"

[keys.normal]
Cmd-c = "yank"
Cmd-v = "paste_after"

[keys.select]
Cmd-c = "yank"
Cmd-v = "paste_after"
```

  - I have also tried the above configuration with numerous other yank [commands](https://docs.helix-editor.com/commands.html): `:yank`, `:yank-join` `:clipboard-yank` `:clipboard-yank-join` `:primary-clipboard-yank` `:primary-clipboard-yank-join` `yank` `yank_joined` `yank_to_clipboard` `yank_to_primary_clipboard` `yank_joined_to_clipboard` `yank_main_selection_to_clipboard` `yank_joined_to_primary_clipboard` `yank_main_selection_to_primary_clipboard` etc
  
Ultimately, `Cmd-c` doesn't yank to the `+` register.
  
I do have another keybinding in the `wezterm` config file which works in the terminal, but not within helix:
  ```
   keys = {
          {
            key = 'c',
            mods = 'CMD',
            action = wezterm.action.CopyTo 'Clipboard',
          },
          {
            key = 'v',
            mods = 'CMD',
            action = wezterm.action.PasteFrom 'Clipboard',
          },
        },
```

I am also running `niri` but have disabled `niri`'s bindings to `cmd+c` and `cmd+v`.  

As far as I know, nothing else should be bound to the command.

Just to be sure, I disabled the `wezterm` key-binding, and still `hx` does not detect the yank.
I should also add, that I've gotten the `Cmd-v` binding to work, but nothing seems to work with `Cmd-c` in particular.
